src/Components/ServiceScene/JSONPanel/highlightLineFilterMatches.ts

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import { AdHocFilterWithLabels } from '@grafana/scenes';
+
+import {
+  getLineFilterRegExps,
+  getMatchingIntervals,
+  highlightValueStringMatches,
+  mergeOverlapping,
+} from '../../../services/highlight';
+
+export function highlightLineFilterMatches(lineFilters: AdHocFilterWithLabels[], value: string, className?: string) {
+  const matchExpressions = getLineFilterRegExps(lineFilters);
+  const lineFilterMatches = getMatchingIntervals(matchExpressions, value);
+  const size = mergeOverlapping(lineFilterMatches);
+  let valueArray: Array<React.JSX.Element | string> = [];
```

Comment:
I see a couple more of usages of this, maybe worth adding a type for `React.JSX.Element | string`

Reply:
https://github.com/grafana/logs-drilldown/pull/1382/commits/fe9417cec671427a8b953c407f0011333f88dac3